test/e2e/framework/job/wait.go

```diff
@@ -27,8 +28,11 @@ import (
 	"k8s.io/apimachinery/pkg/util/wait"
 	clientset "k8s.io/client-go/kubernetes"
 	"k8s.io/kubernetes/test/e2e/framework"
+	"k8s.io/kubernetes/test/utils/format"
 )
 
+type jobCondition func(job *batchv1.Job) (bool, error)
```

Comment:
As users of this packages might have to declare a variable of this type it makes sense to export it. Please also document the semantic of the return values.

Reply:
I know, pod/wait.go doesn't do that either - that's from before my time (4969a05327be), I would have done it differently. But changing it in pod/wait.go would have broken quite a bit of code.

Speaking of differently: the prototype seems to come from the wait package. It allows four different combinations of return values, with some ambiguity what each combination means. Can we get the same effect with a single result, for example a simple failure string? Returning an empty string implies success.

Then the `conditionDesc` parameter in `WaitForJobState` can be dropped.